### ISSUE TITLE
I can't delete chat conversations

### ISSUE THREAD

**Issue by daquinons** [keybase/client#6096]
This can sound silly, but I was testing the chat feature and I didn't want to bother anyone, so I tried to text myself to see how it worked.

But now I'm not able to delete the conversations. I could delete the individual messages, tho

![](http://i.imgur.com/5hpRFmm.png)

Yes, the first conversation is a multichat with myself, myself and me.

**Comment by daquinons** [comment]
After restarting the client the first conversation (the multi one) disappeared but the second one it's still there.

**Comment by ziyaddin** [comment]
Yeah, the same issue. Can not find out how to remove them. Maybe there is no function for it.

**Comment by bw-aus** [comment]
There are two potential issues: **hiding/closing a chat**; and **permanently deleting a chat**.

**Hiding/closing**: it would be good to be able to clean up your sidebar while still keeping chats for reference purposes. Perhaps they could be placed into a folder at the bottom of the sidebar? One that can be expanded when needed?

**Permanently deleting**: could be implemented in a few different ways, one might be just to automate the process of selecting all your own messages in the chat and deleting them, another might be to propose to your chat participants that the entire chat be deleted, and if they all agree then everyone's messages are deleted at the same time and the chat ceases to exist.

I think it is important to implement these features. Some people want to use keybase for conversations that should not last forever. Deleting individual messages is not simple enough.

**Comment by foss-** [comment]
This issue also exists for the iOS / Android variant of keybase, which is expected at this point in time. Just wanted to note, so that once this issue gets addressed, ideally, it will be solved on all platforms.

**Comment by clayzermk1** [comment]
👍 

OS X 10.12.4 (16E195)
Keybase 1.0.23-20170522181119+8a8aea0

Accidentally clicked "start a new chat" on a friend's screen and had two different conversation tabs for them. Quitting and relaunching removed the unused tab.

**Comment by toerict** [comment]
I don't believe keybase has this functionality yet. Hopefully it will be coming soon.

**Comment by nimble-software** [comment]
+1 for being able to delete a chat

**Comment by Oneiroi** [comment]
Being able to at least close an inactive chat would be ideal, e.g. when producing screencasts for training I'd rather not see the last response the other person has sent esp if that response is over a month in age, and closing (or hiding) the conversation would be preferred.

**Comment by miohtama** [comment]
Being not able to delete message or chats greatly decreases the value of Keybase chat as a medium for sharing secrets.

**Comment by eddiewang** [comment]
+1

**Comment by unlocKing** [comment]
Having the same issue, need to be able to delete chats:

https://github.com/keybase/client/issues/8219

**Comment by Leproide** [comment]
Up
Please add chat delete options

**Comment by junderw** [comment]
The lowest hanging fruit should be an "archive" feature that hides certain chats, and should be implemented either way.

The next up would be something like "delete chat" which will have two options. "Ask participants to delete" or "just delete mine"

The next step would be a self destruct timer a la telegram.

The archive feature should be easy enough though... if only I knew go... :-(

**Comment by musl** [comment]
+1 for secure data destruction. IMHO this is an essential part of any secure communications platform.

**Comment by ansoni-san** [comment]
No official response? This is a terrible issue . Will avoid the chat.

**Comment by zuphzuph** [comment]
need.

**Comment by UWTD** [comment]
You can use 
`keybase hide channelname`

**Comment by invious** [comment]
Please implement this!

**Comment by zuphzuph** [comment]
doesn't work in windows keybase shell

**Comment by guycfm** [comment]
I tried that but found that there is no "keybase hide" command!  

We would really like to be able to hide chats that we consider to be inactive.  Maybe build it into the logout/login function so you have a way to start clean.

**Comment by jtx99** [comment]
Any clue when this will be implemented?

It's frustrating to see chat dialog from a YEAR AGO on my screen.  

I wan't to remove these chat windows!

**Comment by zuphzuph** [comment]
Meh, keybase is a dying breed at this point. Sad to see since it had so much promise at one point. 

@jtx99

**Comment by huyz** [comment]
Doubt it’s dying. Did you see how many services they launched last year?

Anyway, you should be able to hide conversations, if not in the client, then at the command line.

**Comment by Leproide** [comment]
@huyz how? And in any case hide != delete/secure delete

**Comment by strib** [comment]
We have both `keybase chat delete-history` and `keybase chat hide`.  (The latter currently has a bug on macOS at least for conversations that have been created since the newest keybase release, but will be fixed soon.)

**Comment by Leproide** [comment]
@strib "ERROR unable to create conversation on finalized TLF"

**Comment by strib** [comment]
That could be a symptom of the bug I mentioned, not sure.  cc: @mmaxim

**Comment by huyz** [comment]
@strib  When did `keybase chat delete-history` get released?  I could swear I just looked at this last-week and didn't see it.

**Comment by strib** [comment]
@huyz within the last month or so.  Released at different times on different platforms, so hard to say exactly.

**Comment by huyz** [comment]
Hmm `keybase chat delete-history` worked but `keybase chat delete-history --age=1w` did not.

**Comment by agsimeonov** [comment]
@strib I too am getting: 
`ERROR unable to create conversation on finalized TLF`.
  
This happens when running:
`keybase chat delete-history'

I should be noted it only happens on a conversation such that one of the members has:
`reset their profile`
`Their encryption keys were replaced with new ones`

More info here:  https://github.com/keybase/keybase-issues/issues/3180

**Comment by strib** [comment]
^ @mmaxim @mlsteele

**Comment by mmaxim** [comment]
Hmm yeah delete history is not going to work on finalized conversations at the moment, we are going to have to think about how to make that work.

**Comment by marcinx** [comment]
Is this on the roadmap? Had a confidential chat with someone and am now unable to delete the chat icon from the list. Very disappointed.

**Comment by mmaxim** [comment]
Did you try `keybase chat hide` on it?

**Comment by marcinx** [comment]
I did and that seems to have done the job. Does this mean my chat partner is blocked though? `-h` seems to use the terms hide/block together. All I want is the history to disappear but make it possible to write me again in the future.

**Comment by mmaxim** [comment]
Conversation is only blocked if you passed in `-b`. The conversation should now not show in your list of conversations, and if you try to write into it again (you can do that by trying to create the conversation again with the other participant (all your old chats will still be there)) it will show up.

**Comment by marcinx** [comment]
Fantastic! Thanks, Mike, I hope this can make its way into the UI at some point. Should be easy enough to show a little X icon in the chats list.

**Comment by Barathwaja** [comment]
Is this feature still open? We badly need this feature 😔

**Comment by the-exodus** [comment]
For now, just type /hide in the chat you want to hide.

**Comment by fooness** [comment]
It’s still not possible to delete chat conversations. Completely empty (cleared) chat conversations are still shown in the list. Hiding a chat conversation is not the same as deleting it. Would be great to finally have this possibility.

**Comment by oblalex** [comment]
I cannot delete locked-out empty conversations with people I even don't know:

![Screen Shot 2019-03-15 at 16 50 25](https://user-images.githubusercontent.com/904705/54440297-1445a400-4743-11e9-86e6-7126e1600ff2.png)
